### PR TITLE
post update to pe, add patch and delete pe in view

### PR DESCRIPTION
**Pull request by shakeh**
order for get still not working i'll make another commit

### CODE REVIEW

File: src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java [hunk 1]
```diff
@@ -39,47 +40,63 @@
...
+            Elements post = new Elements();
+            Element e = new Element();
+            e.put("id", presentationId);
+            if (pe.getName() != null) {
+                e.put("name", pe.getName());
+            }
+            if (pe.getContent() != null) {
+                e.put("documentation", pe.getContent());
+            }
+            if (pe.getType() != null) {
```
Comment (dlamoris) [inline src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java]: need to change classifierIds also if type changed

**Review by dlamoris — CHANGES_REQUESTED** [review]

File: src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java [hunk 2]
```diff
@@ -39,47 +40,63 @@
...
         try {
-            client = Utils.createClient(ticket, auth, url);
-            apiInstance = new ElementApi();
+            ApiClient client = Utils.createClient(ticket, auth, url);
+            ElementApi apiInstance = new ElementApi();
             apiInstance.setApiClient(client);
             Element pe = Utils.getElement(apiInstance, projectId, refId, presentationId, null);
             PresentationElement responsePE = Utils.buildResponsePe(pe);
             response.put("element", responsePE);
+            response.put("status", "ok");
```
Comment (dlamoris) [inline src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java]: change the response to array of "elements" to be consistent

Comment (shakeh) [reply src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java]: what do you want to return? the swagger doesn't have anything for these endpoints... should i just remove the "status"?

File: src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java [hunk 3]
```diff
@@ -39,47 +40,55 @@
...
         try {
-            client = Utils.createClient(ticket, auth, url);
-            apiInstance = new ElementApi();
+            ApiClient client = Utils.createClient(ticket, auth, url);
+            ElementApi apiInstance = new ElementApi();
             apiInstance.setApiClient(client);
             Element pe = Utils.getElement(apiInstance, projectId, refId, presentationId, null);
             PresentationElement responsePE = Utils.buildResponsePe(pe);
-            response.put("element", responsePE);
+            response.put("elements", responsePE);
```
Comment (dlamoris) [inline src/main/java/gov/nasa/jpl/mbee/controllers/PresentationController.java:52]: wrap the responsePE in an array (i updated the swagger to match)

File: src/main/java/gov/nasa/jpl/mbee/controllers/ViewElementsController.java [hunk 4]
```diff
@@ -47,38 +47,33 @@
...
             apiInstance.setApiClient(client);
-            Element view = Utils.getElement(apiInstance, projectId, refId, viewId, null);
 
-//    TODO make util method -  Get child pe ids (need to change to array so it's in order)
-            Set<String> peIds = new HashSet<String>();
-            LinkedTreeMap contents = (LinkedTreeMap) view.get("_contents");
-            ArrayList<Map> pes = (ArrayList) contents.get("operand");
+            Element view = Utils.getElement(apiInstance, projectId, refId, viewId, null);
+            List<String> peIds = new ArrayList<>();
+            Map contents = (Map) view.get("_contents");
```
Comment (dlamoris) [inline src/main/java/gov/nasa/jpl/mbee/controllers/ViewElementsController.java:59]: check for null

**Pull request merged**